plone/app/contenttypes/browser/templates/document.pt

```diff
@@ -12,7 +12,9 @@
                     tal:define="toc python:  getattr(context, 'table_of_contents', False);">
 
   <section id="section-text">
-    <div id="parent-fieldname-text" tal:condition="context/text" tal:content="structure python:context.text.output_relative_to(view.context)" tal:attributes="class python: toc and 'pat-autotoc' or ''">
+    <div id="parent-fieldname-text" class="${python: toc and 'pat-autotoc' or ''}"
+         tal:condition="context/text | nothing"
```

Comment:
```suggestion
         tal:condition="python:getattr(context, 'text', None)"
```

Just tried it out and it seems that the path expression like it used to work it does not anymore... When did the path expressions start throwing an error if the property doesn't exist? From Python 3 on?

Reply:
strange... I just tried it out by changeing this to `context/texter|nothing` and it worked without error ... also disabling richtext behavior on documents and creating a new one throw's no error here ... this should definitely work. I have several `|nothing` conditions in my templates in plone 5.2/py3 ...

Reply:
weird indeed, with kitconcept.volto document ct redefined, without richtext behavior:

```
2021-04-15 15:37:57,237 ERROR   [Zope.SiteErrorLog:22][waitress-0] TypeError: http://localhost:8080/Plone/en/aseases/document_view
Traceback (innermost last):
  Module ZPublisher.WSGIPublisher, line 162, in transaction_pubevents
  Module ZPublisher.WSGIPublisher, line 371, in publish_module
  Module ZPublisher.WSGIPublisher, line 266, in publish
  Module ZPublisher.mapply, line 85, in mapply
  Module ZPublisher.WSGIPublisher, line 63, in call_object
  Module zope.browserpage.simpleviewclass, line 41, in __call__
  Module Products.Five.browser.pagetemplatefile, line 126, in __call__
  Module Products.Five.browser.pagetemplatefile, line 58, in __call__
  Module zope.pagetemplate.pagetemplate, line 133, in pt_render
  Module Products.PageTemplates.engine, line 378, in __call__
  Module z3c.pt.pagetemplate, line 176, in render
  Module chameleon.zpt.template, line 302, in render
  Module chameleon.template, line 214, in render
  Module chameleon.utils, line 53, in raise_with_traceback
  Module chameleon.template, line 192, in render
  Module 23c0222f590f74c31f8298afaf4dbbdc, line 288, in render
  Module a3fe480d318378cc40d0dcdafefaff84, line 860, in render_master
  Module a3fe480d318378cc40d0dcdafefaff84, line 1489, in render_content
  Module 23c0222f590f74c31f8298afaf4dbbdc, line 273, in __fill_content_core
  Module 23c0222f590f74c31f8298afaf4dbbdc, line 125, in render_content_core
  Module zope.tales.expressions, line 250, in __call__
  Module Products.PageTemplates.Expressions, line 196, in _eval
  Module Products.PageTemplates.Expressions, line 126, in render
TypeError: __call__() missing 1 required positional argument: 'value'

 - Expression: "context/text | nothing"
 - Filename:   ... egg/plone/app/contenttypes/browser/templates/document.pt
 - Location:   (line 14: col 21)
 - Source:     tal:condition="context/text | nothing"
                              ^^^^^^^^^^^^^^^^^^^^^^
 - Expression: "provider:plone.abovecontentbody"
 - Filename:   ... egg/Products/CMFPlone/browser/templates/main_template.pt
 - Location:   (line 94: col 70)
 - Source:     ...
                                     ^
 - Expression: "context/main_template/macros/master"
 - Filename:   ... egg/plone/app/contenttypes/browser/templates/document.pt
 - Location:   (line 6: col 21)
 - Source:     ... etal:use-macro="context/main_template/macros/master"
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  template: <Products.Five.browser.pagetemplatefile.ViewPageTemplateFile object at 0x1127a4c10>
               options: {}
               args: ()
               nothing: None
               modules: <Products.PageTemplates.ZRPythonExpr._SecureModuleImporter object at 0x10640e700>
               request: <WSGIRequest, URL=http://localhost:8080/Plone/en/aseases/document_view>
               view: <Products.Five.browser.metaconfigure.SimpleViewClass from /Users/sneridagh/Development/eggs/plone.app.contenttypes-2.2.2-py3.8.egg/plone/app/contenttypes/browser/templates/document.pt object at 0x11174de20>
               context: <FolderishDocument at /Plone/en/aseases>
               views: <Products.Five.browser.pagetemplatefile.ViewMapper object at 0x10f620a00>
               here: <FolderishDocument at /Plone/en/aseases>
               container: <FolderishDocument at /Plone/en/aseases>
               root: <Application at >
               traverse_subpath: []
               user: <PropertiedUser 'admin'>
               default: <DEFAULT>
               repeat: <Products.PageTemplates.engine.RepeatDictWrapper object at 0x112746780>
               loop: {}
               target_language: None
               translate: <function BaseTemplate.render.<locals>.translate at 0x1122d8790>
               macroname: 'master'
               attrs: {}
```

Reply:
(using Volto api buildout)
https://github.com/plone/volto/tree/master/api

Reply:
@sneridagh is right `python:getattr(context, 'text', None)` is better because it checks directly for the `text` attribute while `context/text` apparently checks for a `@@text` view before the `text` attribute.
I believe this change is related to some recent `Chameleon` changes (which AFAIK also should be present in Plone 5.2 latest).

Reply:
fixed that